include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -177,30 +164,58 @@ namespace CUDAWrappers
     struct Data
     {
       /**
-       * Pointer to the quadrature points.
+       * Kokkos::View of the quadrature points.
        */
-      point_type *q_points;
+      Kokkos::View<point_type **, MemorySpace::Default::kokkos_space> q_points;
 
       /**
        * Map the position in the local vector to the position in the global
        * vector.
        */
-      types::global_dof_index *local_to_global;
+      Kokkos::View<types::global_dof_index **,
+                   MemorySpace::Default::kokkos_space>
+        local_to_global;
+
+      /**
+       * Kokkos::View of the inverse Jacobian.
+       */
+      Kokkos::View<Number **[dim][dim], MemorySpace::Default::kokkos_space>
+        inv_jacobian;
 
       /**
-       * Pointer to the inverse Jacobian.
+       * Kokkos::View of the Jacobian times the weights.
        */
-      Number *inv_jacobian;
+      Kokkos::View<Number **, MemorySpace::Default::kokkos_space> JxW;
 
       /**
-       * Pointer to the Jacobian times the weights.
+       * Mask deciding where constraints are set on a given cell.
+       */
+      Kokkos::View<dealii::internal::MatrixFreeFunctions::ConstraintKinds *,
+                   MemorySpace::Default::kokkos_space>
+        constraint_mask;
+
+      /**
+       * Values of the shape functions.
+       */
+      Kokkos::View<Number *, MemorySpace::Default::kokkos_space> shape_values;
```

Comment:
I agree this can be done. Actually almost all the Views (JxW, q_points, local_to_global, etc) are read-only. Would it be OK to wait to do this change? This change makes the code more complicated to read and my understanding is that this is useless since Volta. So I would like to measure the effect before doing this change.